File: app/src/main/java/com/mansuiki/pybot/MainActivity.java
package com.mansuiki.pybot;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.Switch;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (!checkPermission()) {
            Intent intent = new Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS");
            startActivity(intent);
            finish();
        }

        Switch masterSwitch = findViewById(R.id.masterSwitch);
        masterSwitch.setOnCheckedChangeListener(new switchListener());

        Button notiButton = findViewById(R.id.notiButton);
        notiButton.setOnClickListener(new buttonListener());

        BotManager.getManager().setContext(this);
    }

    @Override
    protected void onResume() {
        super.onResume();

        if (BotManager.getManager().isRunning()) {
            Switch masterSwitch = findViewById(R.id.masterSwitch);
            masterSwitch.setChecked(true);
        }
    }


    private boolean checkPermission() {
        String[] list = android.provider.Settings.Secure.getString(getContentResolver(), "enabled_notification_listeners").split(":");
        String me = "com.mansuiki.pybot/com.mansuiki.pybot.KakaoListener";
        for (String next : list) {
            if (me.equals(next)) return true;
        }
        return false;
    }

    class buttonListener implements CompoundButton.OnClickListener {

        @Override
        public void onClick(View view) {

            switch (view.getId()) {

                case R.id.notiButton:
                    Intent intent = new Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS");
                    startActivity(intent);
                    break;
            }
        }
    }

    class switchListener implements CompoundButton.OnCheckedChangeListener {
        @Override
        public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
            if (isChecked) {
                Log.d("Main", "onCheckedChanged: Checked");
                BotManager.getManager().start();
            } else {
                Log.d("Main", "onCheckedChanged: UnChecked");
                BotManager.getManager().stop();
            }

        }
    }
}

File: app/src/main/java/com/mansuiki/pybot/ui/MainActivity.kt
package com.mansuiki.pybot.ui

import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.mansuiki.pybot.ui.background.BotManager
import com.mansuiki.pybot.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        if (!checkPermission()) {
            val intent = Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS")
            startActivity(intent)
            finish()
        }

        binding.masterSwitch.setOnCheckedChangeListener { _, isChecked: Boolean ->
            if (isChecked) {
                Log.d("Main", "onCheckedChanged: Checked")
                BotManager.manager.start()
            } else {
                Log.d("Main", "onCheckedChanged: UnChecked")
                BotManager.manager.stop()
            }
        }

        binding.notiButton.setOnClickListener { _ ->
            val intent = Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS")
            startActivity(intent)
        }

        BotManager.manager.setContext(this)
    }

    override fun onResume() {
        super.onResume()
        if (BotManager.manager.isRunning) {
            binding.masterSwitch.isChecked = true
        }
    }

    private fun checkPermission(): Boolean {
        val list = Settings.Secure.getString(contentResolver, "enabled_notification_listeners").split(":").toTypedArray()
        val me = "com.mansuiki.pybot/com.mansuiki.pybot.ui.background.KakaoListener"
        for (next in list) {
            if (me == next) return true
        }
        return false
    }
}